lib/src/rules/type_init_formals.dart

```diff
@@ -60,7 +60,12 @@ class _Visitor extends SimpleAstVisitor<void> {
   @override
   void visitFieldFormalParameter(FieldFormalParameter node) {
     if (node.type != null) {
-      rule.reportLint(node.type);
+      var class_ =
+          node.thisOrAncestorOfType<ClassDeclaration>()?.declaredElement;
+      var field = class_.getField(node.identifier.name);
+      if (node.type?.type == field.type) {
```

Comment:
Consider assigning `node.type` to a local variable so that we don't need to do the extra null check here.

On the other hand, invalid code might (I haven't tested this) result in `field` being `null` if the field formal parameter references a non-existent field:
```
class C {
  C(int this.x);
}
```

Reply:
Good catch! Caught a crash. Added a test and fixed.